### PR TITLE
Add brainreg recipe

### PR DESCRIPTION
**Pull request by goanpeca**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by goanpeca** [comment]
Depends on:
- https://github.com/conda-forge/staged-recipes/pull/17755
- https://github.com/conda-forge/staged-recipes/pull/17712
- https://github.com/conda-forge/staged-recipes/pull/17754
- https://github.com/conda-forge/staged-recipes/pull/17829
- https://github.com/conda-forge/staged-recipes/pull/17763
- https://github.com/conda-forge/staged-recipes/pull/17820
- https://github.com/conda-forge/staged-recipes/pull/17756
- https://github.com/conda-forge/staged-recipes/pull/17758

**Comment by goanpeca** [comment]
We are in the process of creating conda packages for [napari plugins](https://www.napari-hub.org/) and related dependencies. Once this recipe passes all the checks, every time a new release goes into PyPI, a PR will be created automatically, with usually very minor need for changes so the process of maintaining this moving forward is small. Also the @conda-forge/napari team will be available to help as needed.

You can read a bit more about conda-forge at https://conda-forge.org/#contribute

--- 

@adamltyson, do you agree with being added as a maintainer?

**Comment by adamltyson** [comment]
Yep

**Comment by jaimergp** [comment]
The linking errors are produced by the binary vendoring happening in this package:

```
  adding 'brainreg/bin/nifty_reg/linux_x64/reg_aladin'
  adding 'brainreg/bin/nifty_reg/linux_x64/reg_average'
  adding 'brainreg/bin/nifty_reg/linux_x64/reg_f3d'
  adding 'brainreg/bin/nifty_reg/linux_x64/reg_jacobian'
  adding 'brainreg/bin/nifty_reg/linux_x64/reg_measure'
  adding 'brainreg/bin/nifty_reg/linux_x64/reg_resample'
  adding 'brainreg/bin/nifty_reg/linux_x64/reg_tools'
  adding 'brainreg/bin/nifty_reg/linux_x64/reg_transform'
  adding 'brainreg/bin/nifty_reg/osX/reg_aladin'
  adding 'brainreg/bin/nifty_reg/osX/reg_average'
  adding 'brainreg/bin/nifty_reg/osX/reg_f3d'
  adding 'brainreg/bin/nifty_reg/osX/reg_jacobian'
  adding 'brainreg/bin/nifty_reg/osX/reg_measure'
  adding 'brainreg/bin/nifty_reg/osX/reg_resample'
  adding 'brainreg/bin/nifty_reg/osX/reg_tools'
  adding 'brainreg/bin/nifty_reg/osX/reg_transform'
  adding 'brainreg/bin/nifty_reg/win64/reg_aladin.exe'
  adding 'brainreg/bin/nifty_reg/win64/reg_average.exe'
  adding 'brainreg/bin/nifty_reg/win64/reg_f3d.exe'
  adding 'brainreg/bin/nifty_reg/win64/reg_jacobian.exe'
  adding 'brainreg/bin/nifty_reg/win64/reg_measure.exe'
  adding 'brainreg/bin/nifty_reg/win64/reg_resample.exe'
  adding 'brainreg/bin/nifty_reg/win64/reg_tools.exe'
  adding 'brainreg/bin/nifty_reg/win64/reg_transform.exe'
  adding 'brainreg/bin/nifty_reg/win64/vcruntime140_1.dll'
```

If those binaries need to be distributed on `conda-forge`, they should be compiled on conda-forge, with openly available source code. I am assuming these tools come from [`niftyreg`](https://github.com/KCL-BMEIS/niftyreg), so we would need to build that and then have this tool depend on the package (possibly in a way that does not assume the binaries are contained within the Python source code).

In short:

* We need to package nifty_reg
* Submit a patch upstream to allow other locations

Thoughts @adamltyson? Is this feasible?

**Comment by adamltyson** [comment]
> I am assuming these tools come from [niftyreg](https://github.com/KCL-BMEIS/niftyreg) 

correct

> Thoughts @adamltyson? Is this feasible?

Having `niftyreg` on conda would be great, but I don't know how practical that is. I don't remember having much trouble compiling it, but that's about all I can say.

I'd like the package from PyPI to "just work" without a seperate installation of niftyreg, but I'd be happy to not package the binaries for conda-forge (is it possible to just exclude `bin`?), and then brainreg could have the fallback behaviour of:
- If can't run `reg_aladin` (not compiled locally or installed from conda-forge), then run `brainreg/bin/nifty_reg/linux_x64/reg_aladin`

**Comment by jaimergp** [comment]
Yes, that's a perfectly valid compromise! Thanks! 

We can definitely exclude `bin` from this submission. We'll try to set up the recipe for `niftyreg` in the coming days but we have some other things in the to-do list with higher priority. In the meantime, if you can start the patch to find `reg_alading` (and friends) in `PATH`, that would be awesome!

**Comment by jaimergp** [comment]
Depends on:

* https://github.com/conda-forge/staged-recipes/pull/18531

**Comment by goanpeca** [comment]
Still working on this!

**Comment by adamltyson** [comment]
Hi @goanpeca - is there anything I can do to push this forwards? I'm afraid I don't know enough about conda-forge to assess the status of this PR, but `niftyreg` is now on conda-forge.

**Comment by dstansby** [comment]
👋  @goanpeca do you have any plans to work on this in the near future? If not I would be able to take over this and try and get it finished.

**Comment by dstansby** [comment]
I've taken this and updated it to work in https://github.com/conda-forge/staged-recipes/pull/21972, so I think this PR can be closed.

**Comment by goanpeca** [comment]
Hi @dstansby, sure thing!

We are on hold for recipe building while we finish some other parts around napari.

Cheers!
